### PR TITLE
Search for executables in PATH

### PR DESCRIPTION
**Pull request by ozagordi**
Added a mechanism in python code that searches for executables in the
whole PATH locations, if they are not in src/shorah/bin. This works for
installations obtained with pip install, and should also work with
autotools.

Minor modifications include proper logging arguments and use of double
quotes in docstrings (not yet PEP257 compliant).

### CODE REVIEW

File: src/shorah/amplicon.py [hunk 1]
```diff
@@ -53,12 +53,21 @@
 diri_exe = resource_filename(__name__, 'bin/diri_sampler')
 b2w_exe = resource_filename(__name__, 'bin/b2w')
 
+if not os.path.exists(diri_exe) or not os.path.exists(b2w_exe):
+    for path in os.environ["PATH"].split(os.pathsep):
```
Comment (DrYak) [inline src/shorah/amplicon.py]: Code seems correct and does what you want it to do, but I would try to use some standard library stuff instead manually searching it with handcrafted code.

Comment (ozagordi) [reply src/shorah/amplicon.py]: I was looking for a solution and ended up here: https://stackoverflow.com/questions/377017/test-if-executable-exists-in-python.

Comment (DrYak) [reply src/shorah/amplicon.py]: Apparently, we looked at the exact same StackOverflow, but decided to pick different solution. ( Handcrafted vs Standard library)

**Comment by ozagordi** [comment]
Ok, let me rewrite with `shutil.which`. (I don't usually stop at the first solution, especially when it's old, but the most recent comments did not point to the newer solution below).

**Comment by DrYak** [comment]
Here's my suggestion. 
Fill free to pick it up.

**Comment by DrYak** [comment]
Looks okay to me, feel free to merge if it passes your testing.

**Pull request merged**